### PR TITLE
[core] Remove automated DataGrid performance tests

### LINKED ISSUE #10381: [datagrid] Performance tests posted by mui-bot are inaccurate
### What's the problem? 🤔

The datagrid performance tests run by the datagrid are quite inaccurate. For filtering, they often even show test runs with negative runtime, and I know I optimized filtering a lot but I probably did not make it fast enough to reverse the flow of time.

So I was wondering if we should either remove the tests, or spend time to make them accurate? It would also be way nicer if they'd show the before/after time, rather than just a flat value for the current branch.

In their current state, I'd remove the tests because they're noise that I never look at due to their inaccuracy and that they don't provide a comparison.

cc @mui/xgrid 

image

### PR DESCRIPTION
**Pull request by romgrk**
Closes #10381

The implementation of this performance report is wrong enough that there's no value in keeping it as it is. If we want good numbers, we need to run the benchmarking code along with the benchmarked code, not through a puppeteer connection. And add a before/after comparison.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -279,19 +279,11 @@ jobs:
       - checkout
       - install_js:
           browsers: true
-      - run:
-          name: Run performance tests
-          command: xvfb-run yarn test:performance
```
Comment (cherniavskii) [inline .circleci/config.yml:284]: These scripts could be removed as well:
https://github.com/mui/mui-x/blob/6561be7ac4c5bc83f71d62cc12b2cc4e0e81b70e/package.json#L51-L55

**Review by cherniavskii — APPROVED** [review]
I think I developed a blind eye to these metrics 😅

Do you have ideas for a better performance testing setup with meaningful metrics?

**Comment by romgrk** [comment]
> Do you have ideas for a better performance testing setup with meaningful metrics?

We need to at least run in the same thread as the code being tested. Beyond that it depends what we want to measure.

If we want to measure raw computation speed, then we simply wrap our filtering/sorting/grouping functions with `Date.now()` and be done with it.

If we want to test how fast the grid completes those operations after user input, then we have the start timestamp, and the best way to get the end timestamp would be to use a `MutationObserver` to watch the DOM and know when the operation is really complete (including react rendering).

I think something like the karma tests setup might work.

**Comment by cherniavskii** [comment]
How about scrolling performance?

**Comment by romgrk** [comment]
Hard to say. Scrolling performance is harder to measure, it's not just necessarily about CPU usage, it's also about how the CPU is used and how many frames are skipped. So I couldn't say at the moment.

**Comment by romgrk** [comment]
We need to know how smooth the scrolling is. Recently browsers do scrolling even if the UI thread is blocked on something else, so it shouldn't be an issue. But then we need to measure how often the user sees white areas. But that requires knowledge of the DOM state. But reading the DOM might throw a wrench into React's rendering cycle which runs in sync with browser frames, and if we read the DOM at the wrong moment we might trigger a reflow. We'd need to have access to React rendering internals to know when we can safely read the DOM, but we don't. So yeah I don't know.

**Review by MBilalShafi — APPROVED** [review]
> I think I developed a blind eye to these metrics 😅

Likewise 😄

**Pull request merged**